### PR TITLE
feat(console): user details settings

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Implement "settings" in user details page.

Please notice the following TODOs:

1. Readonly style of TextInput: LOG-1914
2. CodeEditor style: LOG-1915
3. Social connectors: LOG-1917

## Linear Issue Reference

LOG-1706

## Testing

image

@logto-io/eng

### CODE REVIEW

File: packages/console/src/pages/UserDetails/index.tsx [hunk 1]
```diff
@@ -1,32 +1,101 @@
...
+
+    if (!customData) {
+      toast.error(t('user_details.custom_data_invalid'));
+
+      return;
+    }
+
+    const payload: Partial<User> = {
+      name: formData.name || undefined,
+      avatar: formData.avatar || undefined,
```
Comment (demonzoo) [inline packages/console/src/pages/UserDetails/index.tsx:78]: why `|| undefined`? Will `name` and `avatar` be `null` or something else negative?

File: packages/console/src/pages/UserDetails/index.tsx [hunk 2]
```diff
@@ -1,32 +1,101 @@
...
+  });
+
   return (
     <div className={styles.container}>
       <BackLink to="/users">{t('user_details.back_to_users')}</BackLink>
 
       {isLoading && <div>loading</div>}
       {error && <div>{`error occurred: ${error.metadata.code}`}</div>}
-      {data && (
+      {id && data && (
```
Comment (demonzoo) [inline packages/console/src/pages/UserDetails/index.tsx:98]: Would be better if the form is visible even if `id` or `data` is empty

File: packages/console/src/pages/UserDetails/index.tsx [hunk 3]
```diff
@@ -1,32 +1,101 @@
...
+      return;
+    }
+    reset({
+      primaryEmail: data.primaryEmail ?? '',
+      primaryPhone: data.primaryPhone ?? '',
+      username: data.username ?? '',
+      name: data.name ?? '',
+      avatar: data.avatar ?? '',
+      roles: data.roleNames.join(','),
+      customData: JSON.stringify(data.customData, null, 2),
```
Comment (demonzoo) [inline packages/console/src/pages/UserDetails/index.tsx:59]: nit: prefer object destructuring?
```typescript
const {
  primaryEmail = '',
  primaryPhone = '',
  username = '',
  ...
} = data;
```

Comment (demonzoo) [reply packages/console/src/pages/UserDetails/index.tsx:59]: Wait, can't we just directly set undefined to those fields?
```typescript
reset({
  ...data,
  roles: data.roleNames.join(','),
  customData: JSON.stringify(data.customData, null, 2),
})
```

File: packages/console/src/pages/UserDetails/index.tsx [hunk 4]
```diff
@@ -40,7 +109,68 @@ const UserDetails = () => {
...
+                  className={styles.textField}
+                >
+                  <TextInput readOnly {...register('roles')} />
+                </FormField>
+                <FormField
+                  isRequired
+                  title="admin_console.user_details.field_custom_data"
+                  className={styles.textField}
+                >
+                  <CodeEditor height="200px" language="json" value={value} onChange={onChange} />
```
Comment (demonzoo) [inline packages/console/src/pages/UserDetails/index.tsx:160]: Can we omit the unit `px`, and make `json` as the default language?

File: packages/console/src/pages/UserDetails/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/UserDetails/index.tsx:98]: Maybe we can optimized it after "Loading" design is ready?

File: packages/console/src/pages/UserDetails/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/UserDetails/index.tsx]: updated

File: packages/console/src/pages/UserDetails/index.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/pages/UserDetails/index.tsx:59]: Per offline discussion, we will ignore this "null" problem for now, and discuss it in the next meeting. LOG-1928

File: packages/console/src/pages/UserDetails/index.tsx (hunk 2 above)
Comment (demonzoo) [reply packages/console/src/pages/UserDetails/index.tsx:98]: ok

File: packages/console/src/pages/UserDetails/index.tsx (hunk 4 above)
Comment (demonzoo) [reply packages/console/src/pages/UserDetails/index.tsx:160]: ok

**Review by demonzoo — APPROVED** [review]

**Pull request merged**